include/psen_scan_v2/reply_msg_from_scanner.h

```diff
@@ -132,21 +152,29 @@ inline ReplyMsgFromScanner::RawType ReplyMsgFromScanner::toCharArray()
   std::ostringstream os;
 
   uint32_t crc{ calcCRC(*this) };
-  os.write((char*)&crc, sizeof(uint32_t));
-  os.write((char*)&reserved_, sizeof(uint32_t));
-  os.write((char*)&opcode_, sizeof(uint32_t));
-  os.write((char*)&res_code_, sizeof(uint32_t));
 
-  ReplyMsgFromScanner::RawType ret_val{};
+  write<uint32_t>(os, crc);
```

Comment:
You can skip the template type here (and in the following lines). The type is automatically determined by the compiler.